cpp/src/neighbors/cagra_serialize.cuh

```diff
@@ -19,212 +19,36 @@
- * #include <raft/core/resources.hpp>
- * #include <cuvs/neighbors/cagra_serialize.hpp>
- *
- * raft::resources handle;
- *
- * // create a string with a filepath
- * std::string filename("/path/to/index");
- * // create an index with `auto index = raft::cagra::build(...);`
- * raft::cagra::serialize(handle, filename, index);
- * @endcode
- *
- * @tparam T data element type
- * @tparam IdxT type of the indices
- *
- * @param[in] handle the raft handle
- * @param[in] filename the file name for saving the index
- * @param[in] index CAGRA index
- * @param[in] include_dataset Whether or not to write out the dataset to the file.
- *
- */
-template <typename T, typename IdxT>
-void serialize(raft::resources const& handle,
-               const std::string& filename,
-               const index<T, IdxT>& index,
-               bool include_dataset = true)
-{
-  detail::serialize(handle, filename, index, include_dataset);
-}
-
-/**
- * Write the CAGRA built index as a base layer HNSW index to an output stream
- *
- * Experimental, both the API and the serialization format are subject to change.
- *
- * @code{.cpp}
- * #include <raft/core/resources.hpp>
- * #include <cuvs/neighbors/cagra_serialize.hpp>
- *
- * raft::resources handle;
- *
- * // create an output stream
- * std::ostream os(std::cout.rdbuf());
- * // create an index with `auto index = raft::cagra::build(...);`
- * raft::cagra::serialize_to_hnswlib(handle, os, index);
- * @endcode
- *
- * @tparam T data element type
- * @tparam IdxT type of the indices
- *
- * @param[in] handle the raft handle
- * @param[in] os output stream
- * @param[in] index CAGRA index
- *
- */
-template <typename T, typename IdxT>
-void serialize_to_hnswlib(raft::resources const& handle,
-                          std::ostream& os,
-                          const cuvs::neighbors::cagra::index<T, IdxT>& index)
-{
-  detail::serialize_to_hnswlib<T, IdxT>(handle, os, index);
-}
-
-/**
- * Save a CAGRA build index in hnswlib base-layer-only serialized format
- *
- * Experimental, both the API and the serialization format are subject to change.
- *
- * @code{.cpp}
- * #include <raft/core/resources.hpp>
- * #include <cuvs/neighbors/cagra_serialize.hpp>
- *
- * raft::resources handle;
- *
- * // create a string with a filepath
- * std::string filename("/path/to/index");
- * // create an index with `auto index = raft::cagra::build(...);`
- * raft::cagra::serialize_to_hnswlib(handle, filename, index);
- * @endcode
- *
- * @tparam T data element type
- * @tparam IdxT type of the indices
- *
- * @param[in] handle the raft handle
- * @param[in] filename the file name for saving the index
- * @param[in] index CAGRA index
- *
- */
-template <typename T, typename IdxT>
-void serialize_to_hnswlib(raft::resources const& handle,
-                          const std::string& filename,
-                          const cuvs::neighbors::cagra::index<T, IdxT>& index)
-{
-  detail::serialize_to_hnswlib<T, IdxT>(handle, filename, index);
-}
```

Comment:
Where is this function now?

Reply:
This function wasn't being publicly exposed (this is in an internal header file in the src/ directory), and wasn't being called anywhere - and since it was dead code I deleted it.

If we need to publicly expose this we can add that as part of another PR that adds this functionality

Reply:
Can we keep it so we don't have to re-add it later? Very soon, we'll be adding CAGRA+HNSW interop to cuVS

Reply:
re-added in last commit